### DISCUSSION TITLE
Add option to avoid certain roads/streets

### DISCUSSION THREAD

**Discussion by micwoj92** [organicmaps/organicmaps#526] [Ideas]
I really would like option to avoid x road on my route. For example by clicking a part of the road.

**Comment by arnaudvergnet** [comment]
This would be especially useful in case of road works or traffic jam.

**Comment by biodranik** [comment]
Could it be fixed by selecting another road and adding an intermediate stop point there?

**Reply by micwoj92** [reply]
Technically yes, but sometimes you can be driving in area which you don't know, so it would be hard to pick another stop there.

**Reply by arnaudvergnet** [reply]
From my experience it does not work well. A lot of time the algorithm makes the route turn around to take the route you want to avoid again.

**Comment by palzkillm** [comment]
Just upvoted this. I live in a city which has roads I can not use with my car because of a pollution ban. Ideally I would like to be able to click start and end on the map, and save it as "Do not go through this road".
While selecting another road and adding an intermediate stop does work, it's incredibly laborious in this specific area, as there are only one way streets around it.

**Reply by biodranik** [reply]
These roads should be marked somehow in OSM, and then it would be possible to "avoid" them by adding a setting.

**Reply by palzkillm** [reply]
For the situation at Stresemannstraße in Hamburg, Germany, it actually IS marked in OSM:
![image](https://user-images.githubusercontent.com/107746723/183281873-273f0930-9436-4f13-953e-fad1652b1fdd.png)

But this then creates the burden for you to actually implement UI into the app for the user to enter their cars fuel type, and logic for the app to check this. 
And while that is pretty straightforward to do, it just needs to be done, the approach of only avoiding roads for specific criteria creates a bottomless well for you to implement:
I.e. in the area where I live, there is a very narrow and small road full with parked cars, which I dont like to use. And there ARE easy options to avoid this road. However, as it's the shortest way, Navigation forces me through it - and once at the beginning of this road it's basically impossible to not use it, because one way streets in the area would require me to drive a major detour. So it would need to be blocked from Navigation from the beginning, as it requires some pre-planning to be avoided.
But the road itself doesn't provide any criteria that makes sense to block it for - it's just a narrow road with cars parking anywhere.

Creating a way to mark a certain road, and simply add it to a block list, is probably a little more work on the UI and logic part - but the multipurpose of this approach would certainly solve a lot of possible use cases.

**Reply by biodranik** [reply]
Let's start with major cases first.

**Reply by muralito** [reply]
Another use case is to avoid a "danger zone" for example areas, some areas should be avoided at all, some others from sunset to sunrise, etc. These kind of data is not allowed to be uploaded to OSM, as it can be seen as discrimination, but it is important data, especially if you are in a city you didnt know well.

**Comment by ryanprior** [comment]
There's lots of reasons for a person to avoid a road that wouldn't make sense to annotate in OSM, like "I was injured there before," "there's a protest there today," or "it's currently submerged in water." A way to mark a specific road to avoid would be really handy!

I've tried to add a waypoint to get it to avoid the road, but often that doesn't help any- it wants to turn that into a little detour that keeps you on the unwanted road most of the time.

Thanks!

**Comment by mcg1878** [comment]
I'd like to avoid narrow roads, especially where there is legally two way traffic but the road isn't wide enough for two vehicles.

**Reply by biodranik** [reply]
It is feasible to implement, if such roads are marked with some special tag in OpenStreetMap.org

**Reply by ryanprior** [reply]
What we need is the ability to have our own personal overlay that modifies the openstreetmap data. I want to use the narrow road that @mcg1878 doesn't like, and I want to avoid the wide road that other people like, for reasons that are personal, transient and not represented in any shared database. That's the entire point of this feature request.

**Reply by mcg1878** [reply]
Of course many vehicles need to avoid height restrictions too.
